packages/orchestrator/lib/routes/v1/putRecurring.ts

```diff
@@ -7,43 +7,59 @@ import { validateRequest } from '@nangohq/utils';
 const path = '/v1/recurring';
 const method = 'PUT';
 
-export type PostRecurring = Endpoint<{
+export type PutRecurring = Endpoint<{
     Method: typeof method;
     Path: typeof path;
     Body: {
-        scheduleName: string;
-        state: 'STARTED' | 'PAUSED' | 'DELETED';
+        schedule: { name: string; state: 'STARTED' | 'PAUSED' | 'DELETED' } | { name: string; frequencyMs: number };
```

Comment:
We currently don't update frequency and state at the same time so making the endpoint a disjoint union type because it is simpler to implement. We will refactor if updating a schedule requires to change multiple attributes at the same time